patches/minecraft/net/minecraft/tags/ITag.java.patch

```diff
@@ -41,18 +41,37 @@
           jsonobject.add("values", jsonarray);
           return jsonobject;
        }
-@@ -178,6 +189,7 @@
+@@ -152,6 +163,8 @@
+       <T> boolean func_230238_a_(Function<ResourceLocation, ITag<T>> p_230238_1_, Function<ResourceLocation, T> p_230238_2_, Consumer<T> p_230238_3_);
+ 
+       void func_230237_a_(JsonArray p_230237_1_);
++
++      default boolean exists(String type, net.minecraftforge.common.data.ExistingFileHelper existingFileHelper) { return true; }
+    }
+ 
+    public static class ItemEntry implements ITag.ITagEntry {
+@@ -178,6 +191,12 @@
        public String toString() {
           return this.field_232969_a_.toString();
        }
 +      @Override public boolean equals(Object o) { return o == this || (o instanceof ITag.ItemEntry && java.util.Objects.equals(this.field_232969_a_, ((ITag.ItemEntry) o).field_232969_a_)); }
++
++      @Override
++      public boolean exists(String type, net.minecraftforge.common.data.ExistingFileHelper existingFileHelper) {
++         return net.minecraftforge.registries.ForgeRegistries.ITEMS.containsKey(field_232969_a_);
```

Comment:
Pretty sure `ItemEntry` is meant for a single entry, `EntityTypeTags` and `FluidTags` are also `ItemEntry`'s.
The `DataGeneratorTest` uses `BlockTags` but I am guessing it worked on the fact that blocks and items share the same name.

Reply:
That's going to pose a problem then. This method is probably going to move to a static with hardcoded type checks since these 4 entry types are fixed, and I can accept the registry as a parameter there.

Reply:
Thinking about it more, this code is completely unnecessary. An `ItemEntry` will never reach this point because validating the registry contains the entry is handled by `func_232963_b_`. I'm going to push a new version that only cares about `TagEntry` and hardcodes it into `TagsProvider`.